cdk/src/main/java/dev/stratospheric/todoapp/cdk/CognitoStack.java

```diff
@@ -136,37 +128,9 @@ private void createOutputParameters(Environment awsEnvironment) {
       .stringValue(this.userPool.getUserPoolProviderUrl())
       .build();
 
-    // CloudFormation does not expose the UserPoolClient secret, so we can't access it directly with
-    // CDK. As a workaround, we create a custom resource that calls the AWS API to get the secret, and
-    // then store it in the parameter store like the other parameters.
-    // Source: https://github.com/aws/aws-cdk/issues/7225
-
-    AwsSdkCall fetchUserPoolClientMetadata = AwsSdkCall.builder()
-      .region(awsEnvironment.getRegion())
-      .service("CognitoIdentityServiceProvider")
-      .action("describeUserPoolClient")
-      .parameters(Map.of(
-        "UserPoolId", this.userPool.getUserPoolId(),
-        "ClientId", this.userPoolClient.getUserPoolClientId()
-      ))
-      .physicalResourceId(PhysicalResourceId.of(this.userPoolClient.getUserPoolClientId()))
-      .build();
-
-    AwsCustomResource describeUserPoolResource = AwsCustomResource.Builder.create(this, "describeUserPool")
-      .resourceType("Custom::DescribeCognitoUserPoolClient")
-      .installLatestAwsSdk(false)
-      .onUpdate(fetchUserPoolClientMetadata)
-      .onCreate(fetchUserPoolClientMetadata)
-      .policy(AwsCustomResourcePolicy.fromSdkCalls(
-        SdkCallsPolicyOptions.builder()
-          .resources(ANY_RESOURCE)
-          .build()
-      ))
-      .build();
-
-    this.userPoolClientSecret = describeUserPoolResource.getResponseField("UserPoolClient.ClientSecret");
+    this.userPoolClientSecret = this.userPoolClient.getUserPoolClientSecret().unsafeUnwrap();
```

Comment:
I assume you also tried to play around a bit to avoid using the `unsafeUnwrap` which will leak the secret to the synthesized CloudFormation template (which is stored in S3), but I haven't figured out how to create a SSM secret from the `SecretValue`, so I guess we're fine with this. I even assume we previously also "leaked" the secret with the custom AWS API call.